internal/evaluator/conftest_evaluator.go

```diff
@@ -65,6 +65,61 @@ func (c CheckResults) ToConftestResults() []output.CheckResult {
 	return results
 }
 
+// trim removes all failure, warning, success or skipped results that depend on
+// a result reported as failure, warning or skipped. Dependencies are declared
+// by setting the metadata via metadataDependsOn.
+func (c *CheckResults) trim() {
+	// holds codes for all failures, warnings or skipped rules, as a map to ease
+	// the lookup, any rule that depends on a reported code will be removed from
+	// the results
+	reported := map[string]bool{}
+
+	for _, checks := range *c {
+		for _, results := range [][]output.Result{checks.Failures, checks.Warnings, checks.Skipped} {
+			for _, result := range results {
+				if code, ok := result.Metadata[metadataCode].(string); ok {
+					reported[code] = true
+				}
+			}
+		}
+	}
+
+	// helper function inlined for ecapsulation, removes any results that depend
+	// on a reported rule, by code
+	trimOutput := func(what []output.Result) []output.Result {
+		if what == nil {
+			// nil might get passed in, while this would not cause an issue, the
+			// function would return empty array and that would needlessly
+			// change the output
+			return nil
+		}
+
+		// holds leftover results, i.e. the ones that do not depend on a rule
+		// reported as failure, warning or skipped
+		trimmed := make([]output.Result, 0, len(what))
+		for _, result := range what {
+			if dependancy, ok := result.Metadata[metadataDependsOn].([]string); ok {
+				for _, d := range dependancy {
+					if !reported[d] {
+						trimmed = append(trimmed, result)
+					}
+				}
+			} else {
+				trimmed = append(trimmed, result)
+			}
+		}
+
+		return trimmed
+	}
+
+	for i, checks := range *c {
+		(*c)[i].Failures = trimOutput(checks.Failures)
+		(*c)[i].Warnings = trimOutput(checks.Warnings)
+		(*c)[i].Skipped = trimOutput(checks.Skipped)
+		(*c)[i].Successes = trimOutput(checks.Successes)
```

Comment:
Is it obvious to everyone why we can't do `checks.Failures = trimOutput(checks.Failures)` and so on? Maybe add a comment if not..?

Reply:
`c[i]` would be a compile error, see [an example](https://go.dev/play/p/Q5xIBKIxqTh)

Reply:
Ok.